File: university/src/main/java/com/gutorov/university/api/category/CategoryRs.java
package com.gutorov.university.api.category;

import java.util.UUID;

public class CategoryRs {
    public UUID id;
    public String name;

    public CategoryRs() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/java/com/gutorov/university/api/category/CategoryRs.kt
package com.gutorov.university.api.category

import com.gutorov.university.entity.CategoryEntity
import java.util.*
import java.util.stream.StreamSupport

class CategoryRs(
    var id: UUID? = null,
    var name: String? = null
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: CategoryEntity?): CategoryRs {
            return CategoryRs(entity?.id, entity?.name)
        }

        @JvmStatic
        fun fromEntityList(entities: List<CategoryEntity>): List<CategoryRs> {
            return StreamSupport.stream(entities.spliterator(), false)
                .map(CategoryRs::fromEntity)
                .toList();
        }
    }
}
